---
title: "編譯器錯誤 CS0156"
ms.date: 07/20/2015
ms.prod: .net
ms.technology: devlang-csharp
ms.topic: article
f1_keywords: CS0156
helpviewer_keywords: CS0156
ms.assetid: 32026b1b-bcd7-4464-b63f-3b38c00452a6
caps.latest.revision: "9"
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 1a24aa212e8f8c1c8e03cdbe583f4c3f5a2e7de1
ms.sourcegitcommit: bd1ef61f4bb794b25383d3d72e71041a5ced172e
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 10/18/2017
---
# <a name="compiler-error-cs0156"></a>編譯器錯誤 CS0156
最內層 catch 子句中巢狀 finally 子句不允許沒有引數的 throw 陳述式  
  
 沒有任何參數的 [throw](../../csharp/language-reference/keywords/throw.md) 陳述式只會出現在不接受任何參數的 **catch** 子句中。  
  
 如需詳細資訊，請參閱[例外狀況處理陳述式](../../csharp/language-reference/keywords/exception-handling-statements.md)和[例外狀況和例外狀況處理](../../csharp/programming-guide/exceptions/index.md)。  
  
 下列範例會產生 CS0156：  
  
```  
// CS0156.cs  
using System;  
  
namespace MyNamespace  
{  
   public class MyClass2 : Exception  
   {  
   }  
  
   public class MyClass  
   {  
      public static void Main()  
      {  
         try  
         {  
            throw;   // CS0156  
         }  
  
         catch(MyClass2)  
         {  
            throw;   // this throw is valid  
         }  
      }  
   }  
}  
```
